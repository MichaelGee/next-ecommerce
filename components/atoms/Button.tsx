import { Button, Icon } from "atomize";


interface IButtonProps {
  text: string;
  fontSize: string;
  padding: string;
  bg: string;
  shadow: string;
  rounded: string;
  fontFamily: string;
  hoverShadow: string;
  iconPosition: string;
  suffix: Element;
  prefix: Element;
  width: string;
  m: object;
}

export const PrimaryButton = (props: IButtonProps) => {
  const {text, padding, bg, rounded, shadow, hoverShadow, suffix, prefix, fontSize, fontFamily, width, m} = props;




  return (
   <>
    <Button
      bg={bg}
      shadow={shadow}
      hoverShadow={hoverShadow}
      rounded={rounded}
      fontSize={fontSize}
      fontFamily={fontFamily}
      p={padding}
      suffix={suffix}
      prefix={prefix}
      width={width}
      m={m}
    >
      {text}
    </Button>
   </>
  )
}