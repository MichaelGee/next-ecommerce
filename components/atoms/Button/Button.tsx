import { Button } from "atomize";


interface IButtonProps {
  text?: string;
  textSize?: string | object;
  padding?: string | object;
  bg?: string;
  shadow?: string;
  rounded?: string;
  fontFamily?: string;
  hoverShadow?: string;
  iconPosition?: string;
  suffix?: Element;
  prefix?: Element;
  width?: string;
  m?: object;
  textTransform?: string;
  hoverBg?: string;
  onClick?: Function;
}

export const PrimaryButton = (props: IButtonProps) => {
  const {text, padding, bg, rounded, shadow, hoverShadow, suffix, prefix, textSize, fontFamily, width, m, textTransform, hoverBg, onClick} = props;

  return (
   <div id='button'>
    <Button
      bg={bg}
      shadow={shadow}
      hoverShadow={hoverShadow}
      rounded={rounded}
      textSize={textSize}
      fontFamily={fontFamily}
      p={padding}
      suffix={suffix}
      prefix={prefix}
      w={width}
      m={m}
      textTransform={textTransform}
      hoverBg={hoverBg}
      onClick={onClick}
    >
      {text}
    </Button>
   </div>
  )
}