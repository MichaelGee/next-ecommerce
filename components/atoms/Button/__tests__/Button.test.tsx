import '@testing-library/jest-dom'
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { PrimaryButton } from '../Button';

configure({ adapter: new Adapter() });



describe('render button', () => {
    it('matches the snapshot', () => {
        const tree = renderer.create(<PrimaryButton />).toJSON();
        expect(tree).toMatchSnapshot();
    })
    test('check if button component exists', () => {
        const wrapper = shallow(<PrimaryButton />);
        expect(wrapper.find(<button />));
    })
})
