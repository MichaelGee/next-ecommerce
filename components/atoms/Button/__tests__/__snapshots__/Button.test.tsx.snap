// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render button matches the snapshot 1`] = `
<div
  id="button"
>
  <button
    className=""
  />
</div>
`;
